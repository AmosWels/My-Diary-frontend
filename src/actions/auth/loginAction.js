import { API_URLS } from "../../appUrls";
import ACTION_TYPE from "../actionTypes";
import { toast } from "react-toastify";
import axios from "axios";

export const loginAction = data => dispatch => {
  return axios
    .post(API_URLS.LOGIN_USER, data)
    .then(response => {
      toast.success("You have Succefully Logged In", {
        autoClose: 4000,
        hideProgressBar: true
      });
      dispatch({
        type: ACTION_TYPE.LOGIN_ACTION,
        payload: response.data
      });
    })
    .catch(error => {
      toast.error(
        "Wrong Credentials",
        {
          autoClose: 4000,
          hideProgressBar: true
        }
      );
      return error.response;
    });
};
