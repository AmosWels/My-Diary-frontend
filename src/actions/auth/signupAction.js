import { API_URLS } from "../../appUrls";
import ACTION_TYPE from "../actionTypes";
import { toast } from "react-toastify";
import axios from "axios";

export const signupAction = data => dispatch => {
  return axios
    .post(API_URLS.SIGNUP_USER, data, {})
    .then(response => {
      toast.success(response.data.Message, {
        autoClose: 4000,
        hideProgressBar: true
      });
      dispatch({
        type: ACTION_TYPE.SIGNUP_ACTION,
        payload: response.data
      });
    })
    .catch(error => {
      toast.error(
        "Password and username should be > 5, or Username already exists",
        {
          autoClose: 4000,
          hideProgressBar: true
        }
      );
      return error.response;
    });
};
