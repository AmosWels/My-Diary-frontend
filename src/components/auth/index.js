import React, { Component } from "react";
import { Redirect} from "react-router-dom";
import LoginView from "../../containers/auth/loginContainer";
class Index extends Component {
  handleLogin = event => {
    event.preventDefault();
    return <Redirect to="/user/entries" />;
  };
  render() {
    return (
      <div>
        <LoginView />
      </div>
    );
  }
}

export default Index;
