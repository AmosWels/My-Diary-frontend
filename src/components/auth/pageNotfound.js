import React, { Component } from 'react';

class PageNotfound extends Component {
    state = { 
    }
    render() { 
        return ( 
            <h1>Page Not found</h1>
        );
    }
}
 
export default PageNotfound;
