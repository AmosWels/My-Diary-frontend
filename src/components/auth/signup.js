import React, { Component } from "react";
import { Redirect } from "react-router-dom";
import axios from "axios";
// import { NotificationManager } from 'react-notifications';
import "react-notifications/lib/notifications.css";
import { baseurl } from "../../App";

class Signup extends Component {
  constructor() {
    super();
    this.state = {
      signedUp: false
    };
  }

    handleSignup = (e) =>{
      e.preventDefault();
      let password =  e.target.elements.password.value;
      let confirm_password = e.target.elements.psw_repeat.value;
      let url = baseurl + "/api/v1/auth/signup";
      if (password !== confirm_password){
        alert("Passwords dont match");
      } else{
        let user = {
          username: e.target.elements.username.value,
          password: password
        };
        axios.post(url, JSON.stringify(user), {
          headers: {"Content-Type": "application/json"}
        }).then(response => {
          alert("Success :" + response.data.Message);
          this.setState({signedUp: true});
        }).catch(error => {
          alert("Fail :" + error.response.data.Message);
        });
      }
    }

    render() { 
      if (this.state.signedUp) {
        return (<Redirect to=""/>);
      }
      return ( 
        <div>
          <h2 className="centered">Sign Up with My Diary</h2>
          <div  id="call"></div>
          <div className="container">
            <form id="register" method="POST" onSubmit={this.handleSignup}>
              <label htmlFor="username">
                <b>Username</b>
              </label>
              <input type="text" placeholder="Enter Username" name="username" id="username1" required />
              <label htmlFor="password">
                <b>Password</b>
              </label>
              <input type="password" placeholder="Enter Password" name="password" id="password1" required />
              <label htmlFor="psw_repeat">
                <b>Repeat Password</b>
              </label>
              <input type="password" placeholder="Repeat Password" name="psw_repeat" id="password2" required />
              <hr/>
              <button type="submit" id="registerbutton">signup</button>
            </form>
          </div>

          <div className="container signin">
            <p>Already have an account?
              <a href="/">Sign in</a>.</p>
          </div>
        </div>
      );
    }
}
 
export default Signup;
