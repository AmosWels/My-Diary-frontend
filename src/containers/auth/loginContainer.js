import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import "../../static/style.css";
import {
  Button,
  Card,
  CardBody,
  CardGroup,
  Col,
  Container,
  Form,
  Input,
  InputGroup,
  InputGroupAddon,
  InputGroupText,
  Row
} from "reactstrap";

const LoginView = ({ handleLogin, handleChange, username, password }) => {
  return (
    <div>
      <div className="app flex-row align-items-center">
        <Container>
          <Row className="justify-content-center">
            <Col md="8">
              <CardGroup>
                <Card className="p-4">
                  <CardBody>
                    <Form onSubmit={handleLogin}>
                      <h1>Login</h1>
                      <p className="text-muted">Sign In to My Diary</p>
                      <InputGroup className="mb-3">
                        <InputGroupAddon addonType="prepend">
                          <InputGroupText>
                            <i className="icon-user" />
                          </InputGroupText>
                        </InputGroupAddon>
                        <Input
                          type="text"
                          placeholder="Username"
                          autoComplete="username"
                          name="username"
                          onChange={handleChange}
                          value={username}
                          required
                        />
                      </InputGroup>
                      <InputGroup className="mb-4">
                        <InputGroupAddon addonType="prepend">
                          <InputGroupText>
                            <i className="icon-lock" />
                          </InputGroupText>
                        </InputGroupAddon>
                        <Input
                          type="password"
                          placeholder="Password"
                          autoComplete="current-password"
                          name="password"
                          onChange={handleChange}
                          value={password}
                          required
                        />
                      </InputGroup>
                      <Row>
                        <Col xs="6">
                          <Button
                            color="primary"
                            className="px-4"
                            type="submit"
                          >
                            Login
                          </Button>
                        </Col>
                        <Col xs="6" className="text-right">
                          <Button color="link" className="px-0">
                            Forgot password?
                          </Button>
                        </Col>
                      </Row>
                    </Form>
                  </CardBody>
                </Card>
                <Card
                  className="text-white bg-primary py-5 d-md-down-none"
                  style={{ width: "44%" }}
                >
                  <CardBody className="text-center">
                    <div>
                      <h2>Sign up</h2>
                      <p>
                        Access all your thoughts, pen them down, update them and
                        manage them all in one place.. How about that?? Huuh
                      </p>
                      <Link to="/auth/signup">
                        <Button
                          color="primary"
                          className="mt-3"
                          active
                          tabIndex={-1}
                        >
                          Register Now!
                        </Button>
                      </Link>
                    </div>
                  </CardBody>
                </Card>
              </CardGroup>
            </Col>
          </Row>
        </Container>
      </div>
    </div>
  );
};
LoginView.propTypes = {
  handleLogin: PropTypes.func,
  handleChange: PropTypes.func,
  username: PropTypes.string,
  password: PropTypes.string
};

export default LoginView;
