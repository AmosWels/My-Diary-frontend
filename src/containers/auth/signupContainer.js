import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import {
  Button,
  Card,
  CardBody,
  CardGroup,
  Col,
  Container,
  Form,
  Input,
  InputGroup,
  InputGroupAddon,
  InputGroupText,
  Row
} from "reactstrap";

const SignUpView = ({
  handleSignup,
  handleChange,
  username,
  password,
  confirm_password
}) => {
  return (
    <div>
      <div className="app flex-row align-items-center">
        <Container>
          <Row className="justify-content-center">
            <Col md="8">
              <CardGroup>
                <Card className="p-4">
                  <CardBody>
                    <Form onSubmit={handleSignup}>
                      <h1>SignUp</h1>
                      <p className="text-muted">Sign Up with My Diary</p>
                      <InputGroup className="mb-3">
                        <InputGroupAddon addonType="prepend">
                          <InputGroupText>
                            <i className="icon-user" />
                          </InputGroupText>
                        </InputGroupAddon>
                        <Input
                          type="text"
                          placeholder="Username"
                          autoComplete="username"
                          onChange={handleChange}
                          name="username"
                          value={username}
                          required
                        />
                      </InputGroup>
                      <InputGroup className="mb-4">
                        <InputGroupAddon addonType="prepend">
                          <InputGroupText>
                            <i className="icon-lock" />
                          </InputGroupText>
                        </InputGroupAddon>
                        <Input
                          type="password"
                          placeholder="Password"
                          autoComplete="current-password"
                          onChange={handleChange}
                          name="password"
                          value={password}
                          required
                        />
                      </InputGroup>
                      <InputGroup className="mb-4">
                        <InputGroupAddon addonType="prepend">
                          <InputGroupText>
                            <i className="icon-lock" />
                          </InputGroupText>
                        </InputGroupAddon>
                        <Input
                          type="password"
                          placeholder="Repeat Password"
                          autoComplete="current-password"
                          onChange={handleChange}
                          name="confirm_password"
                          value={confirm_password}
                          required
                        />
                      </InputGroup>
                      <Row>
                        <Col xs="6">
                          <Button
                            color="primary"
                            className="px-4"
                            type="submit"
                          >
                            Signup
                          </Button>
                        </Col>
                      </Row>
                    </Form>
                  </CardBody>
                </Card>
                <Card
                  className="text-white bg-primary py-5 d-md-down-none"
                  style={{ width: "44%" }}
                >
                  <CardBody className="text-center">
                    <div>
                      <h2>Login</h2>
                      <p>If you already have an account, proceed to login</p>
                      <Link to="/">
                        <Button
                          color="primary"
                          className="mt-3"
                          active
                          tabIndex={-1}
                        >
                          Login Now!
                        </Button>
                      </Link>
                    </div>
                  </CardBody>
                </Card>
              </CardGroup>
            </Col>
          </Row>
        </Container>
      </div>
    </div>
  );
};

SignUpView.propTypes = {
  handleSignup: PropTypes.func,
  handleChange: PropTypes.func,
  username: PropTypes.string,
  password: PropTypes.string,
  confirm_password: PropTypes.string
};

export default SignUpView;
