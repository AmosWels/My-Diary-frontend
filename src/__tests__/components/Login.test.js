import React from "react";
import { shallow, configure } from "enzyme";
import configureMockStore from "redux-mock-store";
import { IndexTest } from "../../components/auth/index";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

describe("SignupUser component", () => {
  const history = { push: jest.fn() };
  const props = {
    history,
    handleChange: jest.fn(),
    handleLogin: jest.fn(),
    dispatch: jest.fn(),
    username: "",
    password: "",
    confirm_password: "",
    window: { location: "/entries" }
  };
  const getEvent = (name = "", value = "") => ({
    preventDefault: jest.fn(),
    target: {
      name,
      value
    }
  });
  const mockStore = configureMockStore();
  let wrapper;

  beforeEach(() => {
    const store = mockStore({});
    wrapper = shallow(<IndexTest {...props} store={store} />);
  });
  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should set state when handle change is called", () => {
    wrapper.instance().handleChange(getEvent("username", "johnson"));
    expect(wrapper.state().username).toEqual("johnson");
  });

  it("should handle Login when called", () => {
    wrapper.instance().handleLogin(getEvent());
    expect(wrapper.state().username).toEqual("");
  });

  it("should not redirect after login failure", () => {
    const nextprops = {
      userdata: {
        isLoggedIn: false
      }
    };
    wrapper.setProps({ ...nextprops });
    expect(history.push).not.toBeCalled();
  });

  it("test view run component will receive props", () => {
    wrapper.setProps({
      userdata: {
        isLoggedIn: true,
        user: ""
      }
    });
  });
});
