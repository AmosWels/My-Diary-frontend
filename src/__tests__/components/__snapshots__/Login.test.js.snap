// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupUser component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Index
    confirm_password=""
    dispatch={[MockFunction]}
    handleChange={[MockFunction]}
    handleLogin={[MockFunction]}
    history={
      Object {
        "push": [MockFunction],
      }
    }
    password=""
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    username=""
    window={
      Object {
        "location": "/entries",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <LoginView
        confirm_password=""
        dispatch={[MockFunction]}
        handleChange={[Function]}
        handleLogin={[Function]}
        history={
          Object {
            "push": [MockFunction],
          }
        }
        password=""
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
        username=""
        window={
          Object {
            "location": "/entries",
          }
        }
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "confirm_password": "",
        "dispatch": [MockFunction],
        "handleChange": [Function],
        "handleLogin": [Function],
        "history": Object {
          "push": [MockFunction],
        },
        "password": "",
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "username": "",
        "window": Object {
          "location": "/entries",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <LoginView
          confirm_password=""
          dispatch={[MockFunction]}
          handleChange={[Function]}
          handleLogin={[Function]}
          history={
            Object {
              "push": [MockFunction],
            }
          }
          password=""
          store={
            Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            }
          }
          username=""
          window={
            Object {
              "location": "/entries",
            }
          }
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "confirm_password": "",
          "dispatch": [MockFunction],
          "handleChange": [Function],
          "handleLogin": [Function],
          "history": Object {
            "push": [MockFunction],
          },
          "password": "",
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "username": "",
          "window": Object {
            "location": "/entries",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
